src/Pact/Eval.hs

```diff
@@ -1173,7 +1173,16 @@ functionApp fnName funTy mod_ as fnBody docs ai = do
   let body = instantiate (resolveArg ai args') fnBody
       fname = asString fnName
       fa = FunApp ai fname mod_ Defun (funTypes fty) docs
-  guardRecursion fname mod_ $ appCall fa ai args' $ fmap (gas,) $ reduceBody body
+
+  returnVal :: Term Name <-
+    guardRecursion fname mod_ $ appCall fa ai args' $ fmap (gas,) $
+    reduceBody body
```

Comment:
Can we format this a bit differently, I found myself reading it as if it were a do-block with two statements:
```
guardRecursion fname mod_
  $ appCall fa ai args'
  $ fmap (gas,) 
  $ reduceBody body
```

Reply:
Additionally, why the type annotation? `reduceBody` is monomorphic.

Reply:
The type ann should be removed. That's for debugging I believe

Reply:
Reformat: sure.
Type annotation: I sprinkle them in in places where they helped me make sense of things. Happy to remove it.

Reply:
I dropped the type annotation and reformatted this to a single line. Longer than 80 chars but consistent with the rest of the file.

Reply:
I have problem with leaving in type annotations, fwiw.